spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/main/java/org/springframework/cloud/gcp/core/autoconfig/GcpContextAutoConfiguration.java

```diff
@@ -44,14 +50,24 @@
 
 	@Bean
 	@ConditionalOnMissingBean
-	public GoogleCredentials googleCredentials() throws Exception {
+	public CredentialsProvider googleCredentials() throws Exception {
 		if (!StringUtils.isEmpty(this.gcpProperties.getCredentialsLocation())) {
-			return GoogleCredentials.fromStream(
+			return () -> GoogleCredentials.fromStream(
```

Comment:
This must be replaced with the `FixedCredentialsProvider` for better readability.